### ISSUE TITLE
@overrides cause maximum recursion depth exceeded when input argument has Literal

### ISSUE THREAD

**Issue by busFred** [mkorpela/overrides#94]
Below are the code that causes `RecursionError`. The code execute without error if `B.method` is not decorated with `@overrides`:
```python
from abc import ABC, abstractmethod
from overrides import overrides
from typing import Literal

class A(ABC):
    @abstractmethod
    def method(self, str: Literal["max","min"]):
        pass
class B(A):
    @overrides
    def method(self, str: Literal["max","min"]):
        return
```
The reported error message is shown below:
```
---------------------------------------------------------------------------
RecursionError                            Traceback (most recent call last)
Input In [3], in <module>

Input In [3], in B()
      9 class B(A):
     10     @overrides
---> 11     def method(self, str: Literal["max","min"]):
     12         return

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/overrides.py:88, in overrides(method, check_signature, check_at_runtime)
     58 """Decorator to indicate that the decorated method overrides a method in
     59 superclass.
     60 The decorator code is executed while loading class. Using this method
   (...)
     85     docstring from super class
     86 """
     87 if method is not None:
---> 88     return _overrides(method, check_signature, check_at_runtime)
     89 else:
     90     return functools.partial(
     91         overrides,
     92         check_signature=check_signature,
     93         check_at_runtime=check_at_runtime,
     94     )

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/overrides.py:114, in _overrides(method, check_signature, check_at_runtime)
    112             return wrapper  # type: ignore
    113         else:
--> 114             _validate_method(method, super_class, check_signature)
    115             return method
    116 raise TypeError(f"{method.__qualname__}: No super class method found")

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/overrides.py:135, in _validate_method(method, super_class, check_signature)
    129     method.__doc__ = super_method.__doc__
    130 if (
    131     check_signature
    132     and not method.__name__.startswith("__")
    133     and not isinstance(super_method, property)
    134 ):
--> 135     ensure_signature_is_compatible(super_method, method, is_static)

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/signature.py:94, in ensure_signature_is_compatible(super_callable, sub_callable, is_static)
     92 if super_type_hints is not None and sub_type_hints is not None:
     93     ensure_return_type_compatibility(super_type_hints, sub_type_hints, method_name)
---> 94     ensure_all_kwargs_defined_in_sub(
     95         super_sig, sub_sig, super_type_hints, sub_type_hints, is_static, method_name
     96     )
     97     ensure_all_positional_args_defined_in_sub(
     98         super_sig,
     99         sub_sig,
   (...)
    104         method_name,
    105     )
    106 ensure_no_extra_args_in_sub(super_sig, sub_sig, is_static, method_name)

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/signature.py:153, in ensure_all_kwargs_defined_in_sub(super_sig, sub_sig, super_type_hints, sub_type_hints, check_first_parameter, method_name)
    146 elif super_index > sub_index and super_param.kind != Parameter.KEYWORD_ONLY:
    147     raise TypeError(
    148         f"{method_name}: `{name}` is not parameter at index `{super_index}`"
    149     )
    150 elif (
    151     name in super_type_hints
    152     and name in sub_type_hints
--> 153     and not _issubtype(super_type_hints[name], sub_type_hints[name])
    154 ):
    155     raise TypeError(
    156         f"`{method_name}: {name} must be a supertype of `{super_param.annotation}` but is `{sub_param.annotation}`"
    157     )

File ~/python_ws/stat721/lib/python3.8/site-packages/overrides/signature.py:42, in _issubtype(left, right)
     40     return True
     41 try:
---> 42     return issubtype(left, right)
     43 except TypeError:
     44     # Ignore all broken cases
     45     return True

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:428, in issubtype(left, right, forward_refs)
    398 def issubtype(
    399     left: Type, right: Type, forward_refs: typing.Optional[dict] = None,
    400 ) -> typing.Optional[bool]:
    401     """Check that the left argument is a subtype of the right.
    402     For unions, check if the type arguments of the left is a subset of the right.
    403     Also works for nested types including ForwardRefs.
   (...)
    426     ```
    427     """
--> 428     return _is_normal_subtype(normalize(left), normalize(right), forward_refs)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:251, in normalize(type_)
    249     args = _normalize_args(frozenset(args))
    250 else:
--> 251     args = _normalize_args(args)
    252 return NormalizedType(origin, args)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in _normalize_args(tps)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in <genexpr>(.0)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in _normalize_args(tps)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in <genexpr>(.0)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

    [... skipping similar frames: <genexpr> at line 232 (982 times), _normalize_args at line 232 (982 times)]

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in _normalize_args(tps)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:232, in <genexpr>(.0)
    230 def _normalize_args(tps: TypeArgs):
    231     if isinstance(tps, collections.abc.Sequence):
--> 232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):
    234         return frozenset(_normalize_args(type_) for type_ in tps)

File ~/python_ws/stat721/lib/python3.8/site-packages/typing_utils/__init__.py:231, in _normalize_args(tps)
    230 def _normalize_args(tps: TypeArgs):
--> 231     if isinstance(tps, collections.abc.Sequence):
    232         return tuple(_normalize_args(type_) for type_ in tps)
    233     if isinstance(tps, collections.abc.Set):

File /usr/lib/python3.8/abc.py:98, in ABCMeta.__instancecheck__(cls, instance)
     96 def __instancecheck__(cls, instance):
     97     """Override for isinstance(instance, cls)."""
---> 98     return _abc_instancecheck(cls, instance)

File /usr/lib/python3.8/abc.py:102, in ABCMeta.__subclasscheck__(cls, subclass)
    100 def __subclasscheck__(cls, subclass):
    101     """Override for issubclass(subclass, cls)."""
--> 102     return _abc_subclasscheck(cls, subclass)

RecursionError: maximum recursion depth exceeded
```

**Comment by mkorpela** [comment]
Can not reproduce. Tried in Python 3.8.7 and Python 3.9.1. With Overrides 6.1.0 and 3.1.0.

**Comment by bryant1410** [comment]
@mkorpela I can reproduce this issue with that exact code. overrides 6.1.0 installed from Conda. Python 3.9.7.

**Comment by bryant1410** [comment]
The problem is that, at some point, `normalize_args` checks if `tps` is a sequence, which it is (the string "max"), so then it takes the first in the iterable, which is "m". Then, recursively, the same is done, and "m" is taken again. Then again, and so on.

I think an extra guard checking for a string-like type needs to be done.

**Comment by bryant1410** [comment]
Note: `_normalize_args` is called by `normalize` from `typing-utils`. I guess then it's an error on their end...

**Comment by bryant1410** [comment]
Yes, it is. Sorry for the number of messages. See bojiang/typing_utils#10

**Comment by mkorpela** [comment]
Thanks @bryant1410 for looking more into this!

**Comment by mkorpela** [comment]
Waiting for typing_utils update to get this fixed.

**Comment by mkorpela** [comment]
@bryant1410 seems no response from typing_utils. I’ll be forking the project to this repository to get this fixed.

**Comment by mkorpela** [comment]
Added test and detected problem.

**Comment by mkorpela** [comment]
Fix released in 6.2.0
